File: frontend/metaozce/lib/entity/Hotel.dart
class Hotel {
  final int id;
  final String otelAd;
  final double fiyat;
  final String fiyatAraligi;
  final String bolge;
  final double havaAlaninaUzakligi;
  final String denizeUzakligi;
  final String plaj;
  final int iskele;
  final int asansor;
  final int acikRestoran;
  final int kapaliRestoran;
  final int acikHavuz;
  final int kapaliHavuz;
  final int bedenselEngelliOdasi;
  final int bar;
  final int suKaydiragi;
  final int baloSalonu;
  final int kuafor;
  final int otopark;
  final int market;
  final int sauna;
  final int doktor;
  final int beachVoley;
  final int fitness;
  final int canliEglence;
  final int wirelessInternet;
  final int animasyon;
  final int sorf;
  final int parasut;
  final int aracKiralama;
  final int kano;
  final int spa;
  final int masaj;
  final int masaTenisi;
  final int cocukHavuzu;
  final int cocukParki;
  final int alaCarteRestoran;

  Hotel({
    required this.id,
    required this.otelAd,
    required this.fiyat,
    required this.fiyatAraligi,
    required this.bolge,
    required this.havaAlaninaUzakligi,
    required this.denizeUzakligi,
    required this.plaj,
    required this.iskele,
    required this.asansor,
    required this.acikRestoran,
    required this.kapaliRestoran,
    required this.acikHavuz,
    required this.kapaliHavuz,
    required this.bedenselEngelliOdasi,
    required this.bar,
    required this.suKaydiragi,
    required this.baloSalonu,
    required this.kuafor,
    required this.otopark,
    required this.market,
    required this.sauna,
    required this.doktor,
    required this.beachVoley,
    required this.fitness,
    required this.canliEglence,
    required this.wirelessInternet,
    required this.animasyon,
    required this.sorf,
    required this.parasut,
    required this.aracKiralama,
    required this.kano,
    required this.spa,
    required this.masaj,
    required this.masaTenisi,
    required this.cocukHavuzu,
    required this.cocukParki,
    required this.alaCarteRestoran,
  });

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'otelAd': otelAd,
      'fiyat': fiyat,
      'fiyatAraligi': fiyatAraligi,
      'bolge': bolge,
      'havaAlaninaUzakligi': havaAlaninaUzakligi,
      'denizeUzakligi': denizeUzakligi,
      'plaj': plaj,
      'iskele': iskele,
      'asansor': asansor,
      'acikRestoran': acikRestoran,
      'kapaliRestoran': kapaliRestoran,
      'acikHavuz': acikHavuz,
      'kapaliHavuz': kapaliHavuz,
      'bedenselEngelliOdasi': bedenselEngelliOdasi,
      'bar': bar,
      'suKaydiragi': suKaydiragi,
      'baloSalonu': baloSalonu,
      'kuafor': kuafor,
      'otopark': otopark,
      'market': market,
      'sauna': sauna,
      'doktor': doktor,
      'beachVoley': beachVoley,
      'fitness': fitness,
      'canliEglence': canliEglence,
      'wirelessInternet': wirelessInternet,
      'animasyon': animasyon,
      'sorf': sorf,
      'parasut': parasut,
      'aracKiralama': aracKiralama,
      'kano': kano,
      'spa': spa,
      'masaj': masaj,
      'masaTenisi': masaTenisi,
      'cocukHavuzu': cocukHavuzu,
      'cocukParki': cocukParki,
      'alaCarteRestoran': alaCarteRestoran,
    };
  }
}

File: GP/MetaOzce_grad_project-nisa/hotel_gp/src/main/java/com/example/hotel_gp/Entity/Hotel.java
package com.example.hotel_gp.Entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Data;
import jakarta.persistence.Column;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

@Entity
@Data
@Table(name = "otel")
@NoArgsConstructor
@AllArgsConstructor
public class Hotel {
    
    @Id 
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", columnDefinition = "INTEGER")
    private int id;

    @Column(name = "otel_ad", length = 250)
    private String otelAd;

    @Column(name = "fiyat")
    private String fiyat;

    @Column(name = "imageurl")
    private String imageurl;

    @Column(name = "score")
    private String score;

    @Column(name = "fiyat_araligi", length = 250)
    private String fiyat_araligi;

    @Column(name = "bolge", length = 250)
    private String bolge;

    @Column(name = "hava_alanina_uzakligi")
    private String hava_alanina_uzakligi;

    @Column(name = "denize_uzakligi")
    private String denizedenize_uzakligiUzakligi;

    @Column(name = "plaj")
    private String plaj;

    @Column(name = "iskele")
    private int iskele;

    @Column(name = "a_la_carte_restoran")
    private int a_la_carte_restoran;

    @Column(name = "asansor")
    private int asansor;

    @Column(name = "acik_restoran")
    private int acik_restoran;

    @Column(name = "kapali_restoran")
    private int kapali_restoran;

    @Column(name = "acik_havuz")
    private int acik_havuz;

    @Column(name = "kapali_havuz")
    private int kapali_havuz;

    @Column(name = "bedensel_engelli_odasi")
    private int bedensel_engelli_odasi;

    @Column(name = "bar")
    private int bar;

    @Column(name = "su_kaydiragi")
    private int su_kaydiragi;

    @Column(name = "balo_salonu")
    private int balo_salonu;

    @Column(name = "kuafor")
    private int kuafor;

    @Column(name = "otopark")
    private int otopark;

    @Column(name = "market")
    private int market;

    @Column(name = "sauna")
    private int sauna;

    @Column(name = "doktor")
    private int doktor;

    @Column(name = "beach_voley")
    private int beach_voley;

    @Column(name = "fitness")
    private int fitness;

    @Column(name = "canli_eglence")
    private int canli_eglence;

    @Column(name = "wireless_internet")
    private int wireless_internet;

    @Column(name = "animasyon")
    private int animasyon;

    @Column(name = "sorf")
    private int sorf;

    @Column(name = "parasut")
    private int parasut;

    @Column(name = "arac_kiralama")
    private int arac_kiralama;

    @Column(name = "kano")
    private int kano;

    @Column(name = "spa")
    private int spa;

    @Column(name = "masaj")
    private int masaj;

    @Column(name = "masa_tenisi")
    private int masa_tenisi;

    @Column(name = "cocuk_havuzu")
    private int cocuk_havuzu;

    @Column(name = "cocuk_parki")
    private int cocuk_parki;

    @Column(name = "predicted_price")
    private double predicted_price;

    @Column(name = "price_difference")
    private double cprice_difference;

    @Column(name = "fair_price_range")
    private String fair_price_range;

    @Column(name= "fair_price_percentage")
    private double fair_price_percentage;

    


    // Getter ve setter'lar
}
